keanu-project/src/main/java/io/improbable/keanu/tensor/validate/policy/ChangeValueTo.java

```diff
@@ -0,0 +1,30 @@
+package io.improbable.keanu.tensor.validate.policy;
+
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+
+public class ChangeValueTo<U, T extends Tensor<U>> implements TensorValidationPolicy<T> {
+    private final U value;
+
+    // package private - because it's created by the factory method TensorValidationPolicy.changeValueTo
+    ChangeValueTo(U value) {
+        this.value = value;
+    }
+
+    @Override
+    public void handle(T tensor, BooleanTensor result) {
+        if (tensor instanceof DoubleTensor) {
+            Double value = (Double) this.value;
+            DoubleTensor mask = result.not().toDoubleMask();
+            ((DoubleTensor) tensor).setWithMaskInPlace(mask, value);
```

Comment:
Will this be a problem with the `inPlace` if possible contract?

Reply:
initially I had this method return a TENSOR but found I needed to do it in-place to make it work with the existing `inPlace` methods. I don't see any problems - am I missing something?